### PR TITLE
Feat/basic search

### PR DESCRIPTION
**Pull request by mahaomolise92**

### CODE REVIEW

File: JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java [hunk 1]
```diff
@@ -88,9 +86,36 @@ public Receive<Event> actor() {
...
+    private Behavior < Event > eventSearchHandler(EventSearchReq request) {
+        libMPI.startTransaction();
+        HashMap<String, Integer > map = new HashMap < String, Integer > ();
+        HashMap<String, String > fields = new HashMap < String, String > ();
+
+        for (int i = 0; i < (request.search().parameters()).size(); i++) {
+            map.put(request.search().parameters().get(i).field(), distance(request.search().parameters().get(i).distance()));
+            fields.put(request.search().parameters().get(i).field(), request.search().parameters().get(i).value());
+        }
+        var recs = libMPI.search(fields.get("national_id"), fields.get("given_name"), fields.get("family_name"), map);
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java:112]: Note that the fields depend on the implementation, so we need : 
- Check the parameters fields against the fields config and filter only the one we have.
- Pass the parameters as a List to the libMPI.search() function.

File: JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/HttpServer.java [hunk 2]
```diff
@@ -529,6 +558,7 @@ private Route createRoutes(final ActorSystem<Void> actorSystem, final ActorRef<B
                                                         () -> routeMatchesForReviewList(actorSystem, backEnd)),
                                                 path("Document",
                                                         () -> routeDocument(actorSystem, backEnd)),
+                                                path("Search", () -> routeSearch(actorSystem, backEnd)),
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/HttpServer.java:561]:
```suggestion
                                                path("patient", () -> routeSearch(actorSystem, backEnd)),
```

File: JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/HttpServer.java [hunk 3]
```diff
@@ -479,6 +481,33 @@ private Route routeLogout() {
...
+                            if (response.isSuccess()) {
+                                final var eventSearchRsp = response.get();
+                                return complete(
+                                        StatusCodes.OK,
+                                        eventSearchRsp,
+                                        Jackson.marshaller());
+                            } else {
+                                return complete(StatusCodes.IM_A_TEAPOT);
+                            }
+                        }));
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/HttpServer.java:497]:
```suggestion
        return requiredSession(refreshable, sessionTransport, session -> entity(Jackson.unmarshaller(Search.class),
                obj -> onComplete(Search(actorSystem, backEnd, obj),
                        response -> {
                            if (response.isSuccess()) {
                                final var eventSearchRsp = response.get();
                                return complete(
                                        StatusCodes.OK,
                                        eventSearchRsp,
                                        Jackson.marshaller());
                            } else {
                                return complete(StatusCodes.IM_A_TEAPOT);
                            }
                        })));
```
Let's ensure the user is authenticated when performing the request.

**Review by marrouchi — CHANGES_REQUESTED** [review]
Looks like a good start ! Left some comments :)

**Comment by marrouchi** [comment]
@mahaomolise92 Could you please add some description to the PR + a documentation about the endpoint in `documentation/api.md` ?

File: JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/HttpServer.java (hunk 2 above)
Comment (MatthewErispe) [reply JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/HttpServer.java:561]: @mahaomolise92 I also noticed that you created a get endpoint for search. Should this not be a post?

**Pull request closed without merging**